app/models/product.rb

```diff
@@ -0,0 +1,28 @@
+class Product < ApplicationRecord
+  validates :name, presence: true, uniqueness: true
+  validates :price, presence: true
+  validates_numericality_of :price, greater_than: 0
+  validates :quantity, presence: true
+  validates_numericality_of :quantity, greater_than: 0
+
+  belongs_to :merchant, :foreign_key => 'merchant_id'
+
+  has_many :reviews
+  has_many :orderitems
+  has_and_belongs_to_many :categories
+
+  def self.in_stock(id, quantity)
```

Comment:
It seems like this may make more sense as an instance method so you can check the quantity of an existing product.